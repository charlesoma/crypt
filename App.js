import React from 'react';
import { View, Dimensions } from 'react-native';
import Home from './components/Home';
import { Provider } from "react-redux";
import { configureStore } from './store/index'

const store = configureStore();
const win = Dimensions.get('window');

export default function App() {
  return (
    <Provider store={store}>
      <View style={{backgroundColor: '#000', height: win.height}}>
        <Home />
      </View>
    </Provider>
  );
}
