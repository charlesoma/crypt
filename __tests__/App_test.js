import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.useFakeTimers();

describe('<App />', () => {
    it('renders without crashing', () => {
        const rendered = renderer.create(<App />).toJSON();
        expect(rendered).toBeTruthy();
    });
    test('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
