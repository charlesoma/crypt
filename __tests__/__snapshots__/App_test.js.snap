// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "height": 1334,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1B2845",
          "borderBottomColor": "#1B2845",
          "borderBottomWidth": 0.5,
          "elevation": 10,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-around",
          "paddingHorizontal": 10,
          "paddingTop": 20,
        }
      }
    >
      <Image
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 64,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="headerContainer"
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#fff",
              "fontSize": 24,
            }
          }
          text="Crypt"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Crypt
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "elevation": 5,
          "height": "100%",
          "width": "100%",
          "zIndex": 5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#1B2845",
            "borderRadius": 12,
            "display": "flex",
            "height": 80,
            "justifyContent": "center",
            "position": "absolute",
            "top": "35%",
            "width": 80,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#fff"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </View>
</View>
`;
